src/workerd/io/tracer.c++

```diff
@@ -278,8 +312,19 @@ void WorkerTracer::setEventInfo(
   trace->bytesUsed = newSize;
 
   KJ_IF_SOME(writer, maybeTailStreamWriter) {
-    writer->report(
-        context, tracing::Onset(cloneEventInfo(info), tracing::Onset::WorkerInfo{}, kj::none));
+    // Provide WorkerInfo to the streaming tail worker if available. This data is provided when the
+    // WorkerTracer is created, but the actual onset event is the best time to send it.
+    auto workerInfo = tracing::Onset::WorkerInfo{
+      .executionModel = trace->executionModel,
+      .scriptName = trace->scriptName.map([](kj::String& s) { return kj::str(s); }),
```

Comment:
Is there a more elegant way to map a `kj::Maybe<kj::String>`? Also do we have a way to clone capnp readers?

Reply:
We do this in a bunch of places, I think it's a good idea to add this to own-utils as another overload to `mapCopyString`